docs/pages/access-controls/reference.mdx

```diff
@@ -373,3 +374,188 @@ Here is an explanation of the fields used in the `where` and `filter` conditions
 
 Check out our [predicate language](../reference/predicate-language.mdx#scoping-allowdeny-rules-in-role-resources)
 guide for a more in depth explanation of the language.
+
+## Template variables and functions
+
+Teleport provides several template variables and functions to enable more robust access control:
+
+### Internal Variables and Functions
+
+#### `{{email.local(variable)}}`
+
+- **Type**: Function
+- **Description**: Extracts the local part of an email address.
+- **Example**: `{{email.local(alice@example.com)}}` -> `alice`
+
+This function is often paired with an external variable, for example:
+`{{email.local(external.username)}}` for IdPs using email addresses as the username.
+
+#### `{{regexp.match(expression, variable)}}`
+
+- **Type**: Function
+- **Description**: Returns true if the provided string matches the specified
+  pattern.
+- **Example**: `{{regexp.match("^Email: (.*)$", external.raw_email)}}` -> `true`
+  for matching input.
```

Comment:
It would be nice to know exactly where each of these are relevant, I don't think it's clear here. For example, `email.local` can be used in any templated role field like `logins` (the list of all fields that support this is long), while I believe `regexp.match` is only applicable in certain fields like `allow.impersonate.where` and `allow.request.claims_to_roles[].value`. I'm happy to help find where these are all relevant

Reply:
Yes, help would be appreciated! Would you feel like adding that content directly, or discussing with me 1:1?

Reply:
This is a section on role templates, I'd suggest we only document the role template functions here, and not the ones found in `where` and `filter` expressions, which should be part of the `Filter Fields` section immediately above and the label expressions page. Role templates are always surrounded in `{{}}`, others are not, so actually `{{regexp.match(expression, variable)}}` is not something that you would ever see, and it is already documented on the predicate language page.

The only functions available in role templates are `email.local` and `regexp.replace`